backend/src/data/repositories/course/course.repository.ts

```diff
@@ -12,6 +15,15 @@ class Course {
     this.#CourseModel = CourseModel;
   }
 
+  async getAll(): Promise<(CourseM & { vendorKey: VendorKey })[]> {
+    return this.#CourseModel
+      .query()
+      .join('vendors', 'vendors.id', 'courses.vendorId')
+      .select('courses.*', 'vendors.key as vendorKey') as QueryBuilder<
+      CourseM & { vendorKey: VendorKey }
+    >;
+  }
```

Comment:
```suggestion
 getAll(): Promise<(CourseM & { vendorKey: VendorKey })[]> {
    return this.#CourseModel
      .query()
      .join('vendors', 'vendors.id', 'courses.vendorId')
      .select('courses.*', 'vendors.key as vendorKey')
      .castTo<(CourseM & { vendorKey: VendorKey })[]>()
      .execute();
  }
```
castTo - to cast type
execute - to get Promise instead of query builder instance
and I believe we will use the same type `CourseM & { vendorKey: VendorKey }` on FE? then move it to shared

Reply:
In shared we already have 
```
type CourseGetResponseDto = {
  id: number;
  title: string;
  description: string;
  url: string;
  vendorKey: VendorKey;
};
```
I think this type should be enough

Reply:
👍 then use CourseGetResponseDto here